src/components/BTable/BTable.vue

```diff
@@ -0,0 +1,320 @@
+          @click="onColumnHeaderClick(header.key)"
+          :disabled="!header.sort"
+          >
+            {{ header.name }}<img v-if="header.name && header.sort && !sorted[header.key]" class="icon" :src="unsorted" />
+            <i v-else-if="sorted[header.key] === 'desc'" class='arrow ion-android-arrow-dropdown' />
+            <i v-else-if="sorted[header.key] === 'asc'" class='arrow ion-android-arrow-dropup' />
+          </button>
+        </th>
+      </tr>
+      <tr
+        v-for="row in rows"
+        class="b-table-row"
+      >
+        <td
+          v-for="cell in row"
+          class="b-table-cell"
+          v-html='cell'
+        />
+        </td>
+      </tr>
+    </table>
+    <div class="pagination">
+      <div class="pagination-buttons">
+        <div class="results">
+          <p>Results per page: </p>
+        </div>
+        <div class="results">
+          <select 
+            id='rowAmount'
+            @change="onRowAmountChange($event.target.value)"
+          >
+            <option
+              v-for="option in options"
+              :value="option"
+              :selected="option === currentRowsPerPage"
+            >
+              {{ option }}
+            </option>
+          </select>
+        </div>
+        <p class="results">
+          {{ results }}
+        </p>
+        <button
+          id="paginate-back"
+          type="button" 
+          class="pagination-button"
+          @click="onPaginateClick(currentPage-1)"
+          :disabled="currentPage <= 0"
+        >
+          <i class='pagination-icon ion-arrow-left-c' aria-hidden='true'></i>
+        </button>
+        <button
+          id="paginate-forward"
+          type="button"
+          class="pagination-button"
+          @click="onPaginateClick(currentPage+1)"
+          :disabled="((currentPage + 1) * currentRowsPerPage) >= totalResults"
+        >
+          <i class='pagination-icon ion-arrow-right-c' aria-hidden='true'></i>
+        </button>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+
+export default {
+
+  name: 'b-table',
+  props: {
+    /**
+    * The column headers for the table.
+    */
+    headers: {
+      type: Array,
+      required: true
+    },
+    /**
+    * The table data that will be displayed, and array of strings and/or objects.
+    */
+    tableData: {
+      type: Array,
+      required: true
+    },
+    /**
+    * Preload a page if provided.  Defualts to zero if not provided.
+    */
+    pageNumber: {
+      type: Number,
+      required: false
+    },
+    /**
+    * The number of rows to be displayed in the table per page. Defualts to 10 if not provided.
+    */
+    rowsPerPage: {
+      type: Number
+    },
+    /**
+    * The total number of rows for the table.
+    */
+    totalResults: {
+      type: Number,
+      required: true
+    },
+    /**
+    * The different row sizes availablt to the user, will default to [10, 20, 50] if not provided.
```

Comment:
`available` is hates you... I hate it too

Reply:
my typing is hurible